network/network_test.go

```diff
@@ -1011,6 +1013,36 @@ func TestNetwork_calculateLamportClock(t *testing.T) {
 	})
 }
 
+func TestNetwork_checkHealth(t *testing.T) {
+	t.Run("up", func(t *testing.T) {
+		trustStore, _ := core.LoadTrustStore("test/truststore.pem")
```

Comment:
Fatal on error—no ignore.

Reply:
test will panic on nil, truststore is used everywhere, thus will break immediately enough when moved/broken. Omitting errors checks keeps tests readable imo

Reply:
Well no. It is not clear to the reader what that second argument is. I had to look it up to see what is not tested. Tests should inform you about the desired result. Not panic so you have to read the code in order to see what even the cause was.

Reply:
I think it's a trade-off; checking errors in test setup makes the test much less readable, but may cause some extra searching when it goes wrong, but then again a panic caused by a nil dereference always reports a stacktrace.
Second return parameter is 99% of the time returned error, confusion isn't very likely there. Adding error checks doubles the test length.

Without err checking:

```
t.Run("up", func(t *testing.T) {
	trustStore, _ := core.LoadTrustStore("test/truststore.pem")
	certificate, _ := tls.LoadX509KeyPair("test/certificate-and-key.pem", "test/certificate-and-key.pem")
	certificate.Leaf, _ = x509.ParseCertificate(certificate.Certificate[0])
	n := Network{
		trustStore:  trustStore,
		certificate: certificate,
	}

	result := n.CheckHealth()

	assert.Equal(t, core.HealthStatusUp, result["tls"].Status)
})
```

With checking:
```
t.Run("up", func(t *testing.T) {
	trustStore, err := core.LoadTrustStore("test/truststore.pem")
	if err != nil {
		panic(err)
	}
	certificate, err := tls.LoadX509KeyPair("test/certificate-and-key.pem", "test/certificate-and-key.pem")
	if err != nil {
		panic(err)
	}
	certificate.Leaf, err = x509.ParseCertificate(certificate.Certificate[0])
	if err != nil {
		panic(err)
	}
	n := Network{
		trustStore:  trustStore,
		certificate: certificate,
	}

	result := n.CheckHealth()

	assert.Equal(t, core.HealthStatusUp, result["tls"].Status)
})
```

Reply:
No `panic(err)` in tests—do `t.Fatal(err)` at the bare minimum.

If you have good tests you can change code and see what happens. If something break in LoadTrustStore now, you'd have to (1) track the applicable test in a stack trace, (2) read the test code, (3) make the test code log [Fatal] **as it should have done in the first place**, (4) run the test again, (5) read the error and fix the issue, and (6) panic again? 🤡

Reply:
Found a less intrusive way to fix it by using
```
require.NoError(t, err)
```

Fixed for this specific test, but will have to be fixed throughout the code (not in this PR though).